### ISSUE TITLE
Request for pipelines to use Python control flow (if, else, elif). 

### ISSUE THREAD

**Issue by FaustineLi** [kubeflow/pipelines#2311]
It would be nice to have runtime control flow using Python idioms (`if`, `else`, `elif`)  for writing the pipeline instead of resorting to `dsl.Condition`.

**Comment by jessiezcc** [comment]
@FaustineLi, can you explain more about what the end user experience you are looking for? Is this more of syntax request or is there use cases blocked by dsl.Condition?

**Comment by Ark-kun** [comment]
> It would be nice to have runtime control flow using Python idioms (`if`, `else`, `elif`) for writing the pipeline instead of resorting to `dsl.Condition`.

At this moment this is non-trivial since the pipeline function is only executed in compile time, not run time. There is no code-executing container task associated with the pipeline which could run the control flow code. Another problem is that with python you have shared variable scope available to all expression, while at runtime you have distributed pods and every piece of data needs to be explicitly passed around.

There are ways to try to analyze the control flow statements in python code, but that would require significant changes to the compilation process.

On the other hand, you can use python idioms like loops or conditional expressions for compile-time code generation.
